registry/confdb_control.go

```diff
@@ -0,0 +1,141 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package registry
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+var (
+	validAccountID = regexp.MustCompile("^(?:[a-z0-9A-Z]{32}|[-a-z0-9]{2,28})$")
+)
+
+type AuthenticationMethod string
+
+const (
+	OperatorKey AuthenticationMethod = "operator-key"
+	Store       AuthenticationMethod = "store"
+)
+
+// isValidAuthenticationMethod checks if a string is a valid AuthenticationMethod
+func isValidAuthenticationMethod(value string) bool {
+	switch AuthenticationMethod(value) {
+	case OperatorKey, Store:
+		return true
+	default:
+		return false
+	}
+}
+
+// convertToAuthenticationMethod converts and validates a []string to []AuthenticationMethod
+func convertToAuthenticationMethod(methods []string) ([]AuthenticationMethod, error) {
+	sort.Slice(methods, func(i, j int) bool {
+		return methods[i] < methods[j]
+	})
+
+	// remove duplicates
+	methods = compact(methods)
```

Comment:
I don't see code in the tests that check that this is done

Reply:
It's a bit hidden but [this test](https://github.com/canonical/snapd/blob/15324821bcb5d36d5caa522faaf97df2f338d0bf/registry/confdb_control_test.go#L39) checks that behavior